xmppserver/src/main/java/org/jivesoftware/openfire/net/SASLAuthentication.java

```diff
@@ -261,8 +261,10 @@ public static Element getSASLMechanismsElement( LocalIncomingServerSession sessi
             if (trusted != null && session.getDefaultIdentity() != null) {
                 haveTrustedCertificate = verifyCertificate(trusted, session.getDefaultIdentity());
             }
-            if (haveTrustedCertificate) {
+            boolean acceptSelfSignedCerts = JiveGlobals.getBooleanProperty(ConnectionSettings.Server.TLS_ACCEPT_SELFSIGNED_CERTS);
```

Comment:
If I read this right, then the `EXTERNAL` mechanism will always be offered when the 'accept self signed' option is enabled.

It would be good to ensure that the certificate is indeed self-signed, before offering that mechanism. I think there's existing code that can be used for that, in `CertificateManager.isSelfSignedCertificate()`

Reply:
The client / initiating server certificate is not sent over and so cannot be evaluated.